src/components/inputs/rich-text-input/rich-text-input.js

```diff
@@ -78,41 +81,65 @@ const plugins = [
   ListPlugin(),
 ];
 
-const RichTextInput = props => {
-  const { onChange } = props;
-  const onValueChange = ({ value }) => {
+class RichTextInput extends React.Component {
+  onValueChange = ({ value }) => {
     const event = {
-      target: { id: props.id, name: props.name, value },
+      target: {
+        id: this.props.id,
+        name: this.props.name,
+        value,
+      },
     };
-    onChange(event);
+    this.props.onChange(event);
   };
 
-  return (
-    <Editor
-      {...filterDataAttributes(props)}
-      id={props.id}
-      name={props.name}
-      disabled={props.isDisabled}
-      readOnly={props.isReadOnly}
-      value={props.value}
-      onFocus={props.onFocus}
-      onBlur={props.onBlur}
-      // we can only pass props to the Editor that Slate understands without getting
-      // warning in the console,
-      // so instead we pass our extra props through this `options` prop.
-      options={{
-        horizontalConstraint: props.horizontalConstraint,
-        defaultExpandMultilineText: props.defaultExpandMultilineText,
-        hasWarning: props.hasWarning,
-        hasError: props.hasError,
-        placeholder: props.placeholder,
-      }}
-      onChange={onValueChange}
-      plugins={plugins}
-      renderEditor={renderEditor}
-    />
-  );
-};
+  shouldComponentUpdate(nextProps) {
+    let remainingPropsHaveChanged = false;
+    const remainingProps = omit(this.props, propsToOmit);
+
+    // check remaining props normally using !==
+    Object.entries(remainingProps).forEach(([key, val]) => {
```

Comment:
this is just how `shouldComponentUpdate` usually works. However, we cannot rely on it for slate's value.